src/features/edge-intersection/sweepline.ts

```diff
@@ -0,0 +1,120 @@
+/********************************************************************************
+ * Copyright (c) 2019 Rowan Winsemius and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+// Based on the sweepline implementation at https://github.com/rowanwins/sweepline-intersections
+// which is published under the terms of MIT, but has been adapted to the use case of sprotty.
+import TinyQueue from "tinyqueue";
+import { Point, PointToPointLine } from "../../utils/geometry";
+import { Intersection } from "./intersection-finder";
+import { RoutedPoint } from "../routing/routing";
+
```

Comment:
Please add comments to the algorithms in this file explaining the basic idea and with links to further information.

I know there's a lot of poorly commented code in this repository, but we need to start improving at some point :-)

Reply:
Sure! I'll add some hopefully helpful doc.